.github/PULL_REQUEST_TEMPLATE.md

```diff
@@ -0,0 +1,38 @@
+<!-- markdownlint-disable MD041 MD043 -->
+<!-- Add issue number next to close keyword so it links PR and issue for auto-closure -->
+closes #ISSUE_NUMBER
+
+## Summary
+
+> Please provide a summary of what's being changed
+
+### User experience
+
+> If applicable, please share what the user experience looks like before and after this change
```

Comment:
i dont see the entire ci/cd process running in this PR . I tried to run it on the cookbook and it fails, the make dev part doesnt work in GitHub

Reply:
got a link with the failure so I can dig into it? I've just tried to enable workflow dispatch (manual trigger on any branch) but it won't work until it merges it. 

If you'd like, I can send a PR only with workflow dispatch enabled in the branch, then I should be able to troubleshoot here before merging this larger PR

Reply:
found the issues, looking into them. It didn't run in this PR because there was a previous run unapproved.

Reply:
Should be operational now. 

The additional problem on why the pipeline fails in unit tests is because an unit test is instantiating a DynamoDB resource client and making a TCP socket with DynamoDB service -- ideally we shouldn't allow that in unit test but still we have to agree on testing conventions.

For now, I'm gonna move the AWS config creds before all tests run to make the pipeline operational, then we can revisit whether we want to strictly disallow any TCP sockets in Unit tests (e.g., security, reproducibility)

<img width="1920" alt="image" src="https://github.com/ran-isenberg/serverless-python-demo/assets/3340292/8e27c52b-27c7-4265-b4a4-da29b6405151">